pyro/infer/autoguide/guides.py

```diff
@@ -1190,3 +1188,280 @@ def forward(self, *args, **kwargs):
+            with pyro.plate("plate", len(data)):
+                y = pyro.sample("y", dist.Normal(0, 1))
+                pyro.sample("z", dist.Normal(y, x), obs=data)
+
+        guide = AutoStructured(
+            model=model,
+            conditionals={"x": "normal", "y": "normal"},
+            dependencies={"x": {"y": "linear"}},
+        )
+
+    Once trained, this guide can be used with
+    :class:`~pyro.infer.reparam.structured.StructuredReparam` to precondition a
+    model for use in HMC and NUTS inference.
+
+    .. note:: If you declare a dependency of a high-dimensional downstream
+        variable on a low-dimensional upstream variable, you may want to use
+        a lower learning rate for that weight, e.g.::
+
+            def optim_config(param_name):
+                config = {"lr": 0.01}
+                if "deps.my_downstream.my_upstream" in param_name:
+                    config["lr"] *= 0.1
+                return config
+
+            adam = pyro.optim.Adam(optim_config)
+
+    :param callable model: A Pyro model.
+    :param conditionals: Family of distribution with which to model each latent
+        variable's conditional posterior. This should be a dict mapping each
+        latent variable name to either a string in ("delta", "normal", or
+        "mvn") or to a callable that returns a sample from a zero mean (or
+        approximately centered) noise distribution (such callables typically
+        call ``pyro.param()`` and ``pyro.sample()`` internally).
+    :param dependencies: Dict mapping each site name to a dict of its upstream
+        dependencies; each inner dict maps upstream site name to either the
+        string "linear" or a callable that maps a *flattened* upstream
+        perturbation to *flattened* downstream perturbation. The string
+        "linear" is equivalent to ``nn.Linear(upstream.numel(),
+        downstream.numel(), bias=False)``.  Dependencies must not contain
+        cycles or self-loops.
+    :param callable init_loc_fn: A per-site initialization function.
+        See :ref:`autoguide-initialization` section for available functions.
+    :param float init_scale: Initial scale for the standard deviation of each
+        (unconstrained transformed) latent variable.
+    :param callable create_plates: An optional function inputing the same
+        ``*args,**kwargs`` as ``model()`` and returning a :class:`pyro.plate`
+        or iterable of plates. Plates not returned will be created
+        automatically as usual. This is useful for data subsampling.
+    """
+
+    scale_constraint = constraints.softplus_positive
+    scale_tril_constraint = constraints.softplus_lower_cholesky
+
+    def __init__(
+        self,
+        model,
+        *,
+        conditionals: Dict[str, Union[str, Callable]] = "normal",
+        dependencies: Dict[str, Dict[str, Union[str, Callable]]] = "linear",
+        init_loc_fn=init_to_feasible,
+        init_scale=0.01,
```

Comment:
This is different from other guides. Do we need a change here?

Reply:
Done. :+1: for consistency